app/views/documents/index.html.haml

```diff
@@ -88,9 +87,6 @@
 :javascript
   $("#category").select2();
 
-  /* $("#category").select2({
-    initSelection: function (elm, callback) {
-      callback({id: elm.data('initValue')});
-    },
-    data: null
-  }); */
+  preselected_values = #{params[:q][:with_all_tags] || []};
+
+  $('#category').select2('val', preselected_values);
```

Comment:
see https://github.com/lksv/basepack/blob/master/app/assets/javascripts/basepack/forms.coffee#L127-L234 (maybe too complex example for this case)